test/is_palindrome_test.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+  Copyright (c) Alexander Zaitsev <zamazan4ik@gmail.com>, 2016
+
+  Distributed under the Boost Software License, Version 1.0. (See
+  accompanying file LICENSE_1_0.txt or copy at
+  http://www.boost.org/LICENSE_1_0.txt)
+
+  See http://www.boost.org/ for latest version.
+*/
+
+#include <boost/config.hpp>
+#include <boost/algorithm/is_palindrome.hpp>
+
+#define BOOST_TEST_MAIN
+#include <boost/test/unit_test.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <list>
+#include <vector>
+
+
+namespace ba = boost::algorithm;
+
+
+template <typename T>
+bool funcComparator(const T& v1, const T& v2)
+{
+    return v1 == v2;
+}
+
+struct functorComparator
+{
+    template <typename T>
+    bool operator()(const T& v1, const T& v2) const
+    {
+        return v1 == v2;
+    }
+};
+
+
+void test_is_palindrome()
+{
+    const std::list<int> empty;
+    const std::vector<char> singleElement{'z'};
```

Comment:
If you write this as `const std::vector<char> singleElement(1, 'z');` then it works on C++03 as well.